metricflow/cli/cli_context.py

```diff
@@ -113,6 +123,24 @@ def model_path_is_for_dbt(self) -> bool:  # noqa: D
 
         return self._model_path_is_for_dbt
 
+    @property
+    def dbt_cloud_job_args(self) -> Optional[DbtCloudJobArgs]:  # noqa: D
+        if self._dbt_cloud_job_args is None:
+            job_id = self.config.get_value(key=CONFIG_DBT_CLOUD_JOB_ID) or ""
+            service_token = self.config.get_value(key=CONFIG_DBT_CLOUD_SERVICE_TOKEN) or ""
+            # If one of them is set, that means there is at least a partial dbt cloud setup
+            if job_id != "" or service_token != "":
+                # now we assert them both, because if one is missing, we want to give an appropriate error
+                assert (
+                    job_id != ""
+                ), f"Incomplete dbt cloud config detected. The config `{CONFIG_DBT_CLOUD_JOB_ID}` was not set."
+                assert (
+                    service_token != ""
+                ), f"Incomplete dbt cloud config detected. The config `{CONFIG_DBT_CLOUD_SERVICE_TOKEN}` was not set."
```

Comment:
I remember putting a lot more into `DbtCloudJobArgs` and having to walk it back... But I don't remember why. I'll give it a shot, hopefully I'm misremembering. If I'm not misremembering, I'll document here what the issue is.

Reply:
**TLDR:** I originally had a `DbtCloudJobArgs` but this proved problematic due to optional dependency import protections. I think this is out of scope for this PR.

-----

Basically it makes sense to have `DbtCloudArgs`. What we really need is a MetricFlow Config class, which we could then do a `is_dbt_cloud()`, `is_dbt_core()`, etc. Currently the closest thing we have is `CLIContext`, but as the name implies its for use with the CLI, if something else wants to instantiate a `MetricFlowEngine`, it likely shouldn't be using the CLIContext. Without a MetricFlow Config class, the next best place to put `DbtCloudArgs` would be in `dbt_cloud_to_model.py`, but then it can't actually be imported anywhere because `dbt_cloud_to_model.py` contains dangerous imports which are optional (and thus imported in a protected manner, and we'd want import outside those guards). I could put it in a separate file all by itself, but the place which it should likely be put in is `metricflow/configuration`, and at that point we're talking about building a MetricFlow Config class (and that should be it's own PR).

Reply:
Ooof, I thought it'd be easy..... shows what I know.

That kind of mishmash of dependency issues often indicates a need for broader restructuring, so I agree it's a whole other task.